### PR TITLE
Make gridbuilder compatible to schema upgrades

### PR DESCRIPTION
**Pull request by margrietpalm**
These changes are mainly tested in https://github.com/nens/schema_300_integration_tests. All integration tests pass, indicating that identical gridadmin files are reproduced.

### CODE REVIEW

File: threedigrid_builder/base/settings.py [hunk 1]
```diff
@@ -39,6 +44,10 @@ class GridSettings:
     @classmethod
     def from_dict(cls, dct):
         """Construct skipping unknown fields and None values"""
+        schema_to_builder_map = {'minimum_cell_size': 'grid_space',
+                                 'calculation_point_distance_1d': 'dist_calc_points',
+                                 'nr_grid_levels': 'kmax'}
+        dct = replace_keys(copy.copy(dct), schema_to_builder_map)
```
Comment (margrietpalm) [inline threedigrid_builder/base/settings.py:50]: In discussion with @daanvaningen we decided to not change the variable names downstream. Therefore this, ugly, mapping is needed. I am slightly worried that more of these are needed in the future creating a monstrosity. So I'm open to suggestions, and we can always fix this later on in the schema upgrade is a monster indeed appears.

File: threedigrid_builder/base/settings.py [hunk 2]
```diff
@@ -150,6 +160,22 @@ def __post_init__(self):
...
+                                 "max_infiltration_volume": "max_infiltration_capacity",
+                                 "max_infiltration_volume_type": "max_infiltration_capacity_type",
+                                 "manhole_aboveground_storage_area": "manhole_storage_area",
+                                 "friction_coefficient": "frict_coef",
+                                 "minimum_table_step_size": "table_step_size",
+                                 "friction_type": "frict_type",
+                                 "friction_coefficient_type": "frict_coef_type",
+                                 "interception": "interception_global",
+                                 }
+        dct = replace_keys(copy.copy(dct), schema_to_builder_map)
```
Comment (margrietpalm) [inline threedigrid_builder/base/settings.py:178]: see https://github.com/nens/threedigrid-builder/pull/356/files#r1555399049

File: threedigrid_builder/interface/db.py [hunk 3]
```diff
@@ -207,17 +203,37 @@ def get_settings(self) -> dict:
...
-                infiltration, "max_infiltration_capacity", default=NO_AGG
+                infiltration, "max_infiltration_volume", default=NO_AGG
             )
-
         if groundwater:
             # default is what the user supplied (MIN/MAX/AVERAGE)
-            _set_initialization_type(groundwater, "groundwater_impervious_layer_level")
-            _set_initialization_type(groundwater, "phreatic_storage_capacity")
+
+            _set_initialization_type(
```
Comment (caspervdw) [inline threedigrid_builder/interface/db.py:211]: This method `_set_initialization_type`, I think you now always supply `type_field` to it? Maybe good to remove the string logic over there accordingly.

**Review by caspervdw — APPROVED** [review]
Some small comments, good that changes are rather limited here.

File: threedigrid_builder/interface/db.py [hunk 4]
```diff
@@ -145,56 +144,53 @@ def get_settings(self) -> dict:
...
+            if model_settings.use_simple_infiltration:
                 infiltration = _object_as_dict(
-                    session.query(models.SimpleInfiltration)
-                    .filter_by(id=global_.simple_infiltration_settings_id)
-                    .one()
+                    session.query(models.SimpleInfiltration).one()
                 )
                 # older sqlites have no max_infiltration_capacity field
-                infiltration.setdefault("max_infiltration_capacity", None)
+                infiltration.setdefault("max_infiltration_volume", None)
```
[the remark this answers is not in the document]
Comment (margrietpalm) [reply threedigrid_builder/interface/db.py:164]: I think that decision is for the PO to make; so let's tag @leendertvanwolfswinkel

[the remark this answers is not in the document]
Comment (margrietpalm) [reply threedigrid_builder/interface/db.py:164]: I'm not following this. To my knowledge `max_infiltration_volume` is not forced to be present in the sqlite. When it's absent, this line is necessary to have the `max_infiltration_volume` field in the `infiltration` dict.

File: threedigrid_builder/interface/db.py (hunk 3 above)
Comment (margrietpalm) [reply threedigrid_builder/interface/db.py:211]: No, there in most cases the `type_field` is not specified. I'll keep an eye on this in further migrations and remove the logic when it's no longer (or almost no longer) in use.

File: threedigrid_builder/interface/db.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (caspervdw) [reply threedigrid_builder/interface/db.py:164]: Any SQLite with version >= 210 has the `max_infiltration_volume` column, see here https://github.com/nens/threedi-schema/blob/6c62a5a98616ebdeae525348d3801ea2751fdc06/threedi_schema/migrations/versions/0210_global_raster_values.py#L33

[the remark this answers is not in the document]
Comment (margrietpalm) [reply threedigrid_builder/interface/db.py]: Ok, removed lines in bdc83da702a2cf7e14846ddf5734bd72139d093e

**Pull request merged**